### PR TITLE
Manage event response in show and index action to be able to redirect

### PR DESCRIPTION
**Pull request by maximehuran**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | o
| Related tickets | mentioned in https://github.com/Sylius/Sylius/issues/10286
| License         | MIT

### CODE REVIEW

**Review by jacquesbh — APPROVED** [review]

**Review by lchrusciel — COMMENTED** [review]
Hey Maxime 👋 

thanks for the contribution. Can you also provide some tests for this behavior? The best combo would be to have Spec + functional test in PHPUnit

**Comment by maximehuran** [comment]
Hi, I added some PHP Spec tests. I am not sure about it but I don't know how to setup the test application for this plugin.
I don't see PHP Unit tests actually managing some redirection on this plugin.

**Comment by loic425** [comment]
> Hi, I added some PHP Spec tests. I am not sure about it but I don't know how to setup the test application for this plugin.
> I don't see PHP Unit tests actually managing some redirection on this plugin.

Juste look at the CI to see how it is set up :)

**Comment by maximehuran** [comment]
I would like to run the CI to check my tests, I think it's the easiest solution

**Comment by maximehuran** [comment]
ping @lchrusciel I added PHP Spec tests :)

**Review by Zales0123 — APPROVED** [review]
Hello Maxime 🖖 I've rebased your PR, let's see how the build passes, but it's good to go in my opinion :)

File: src/Bundle/spec/Controller/ResourceControllerSpec.php [hunk 1]
```diff
@@ -187,6 +187,39 @@ function it_returns_a_response_for_html_view_of_a_single_resource(
...
+
+        $authorizationChecker->isGranted($configuration, 'sylius.product.show')->willReturn(true);
+        $singleResourceProvider->get($configuration, $repository)->willReturn($resource);
+
+        $eventDispatcher->dispatch(ResourceActions::SHOW, $configuration, $resource)->willReturn($event);
+
+        $event->getResponse()->willReturn($response);
+
+        $configuration->isHtmlRequest()->shouldNotBeCalled();
+        $this->createRestView($configuration, $resource)->shouldNotBeCalled();
```
Comment (Zales0123) [inline src/Bundle/spec/Controller/ResourceControllerSpec.php:218]: Oh, ok, this is the problem 💃 `createRestView` is a protected function inside `ResourceController`, so we should rather check if `$this->viewHandler` is called or not 🖖

Comment (Zales0123) [reply src/Bundle/spec/Controller/ResourceControllerSpec.php]: I've allowed myself to fix it :) 🚀

**Comment by Zales0123** [comment]
Thanks, Maxime! :tada:

File: src/Bundle/spec/Controller/ResourceControllerSpec.php (hunk 1 above)
Comment (maximehuran) [reply src/Bundle/spec/Controller/ResourceControllerSpec.php]: Thank you Mateusz you're the king !

**Pull request merged**